crates/polars-time/src/windows/test.rs

```diff
@@ -340,9 +344,10 @@ fn test_boundaries() {
         false,
         Default::default(),
     );
-    assert_eq!(groups[0], [1, 2]); // 00:00:00 -> 00:30:00
-    assert_eq!(groups[1], [3, 2]); // 01:00:00 -> 01:30:00
-    assert_eq!(groups[2], [5, 2]); // 02:00:00 -> 02:30:00
+    assert_eq!(groups[0], [0, 1]); // (2021-12-15 23:30, 2021-12-16 00:00]
+    assert_eq!(groups[1], [1, 2]); // (2021-12-16 00:00, 2021-12-16 00:30]
+    assert_eq!(groups[2], [3, 2]); // (2021-12-16 00:30, 2021-12-16 01:00]
+    assert_eq!(groups[3], [5, 2]); // (2021-12-16 01:00, 2021-12-16 01:30]
```

Comment:
here's an example of where the earliest datapoint is currently being excluded

because `closed='right'`, the `00:00:00` point isn't currently included in any windows. This PR fixes that